File: Java/TMP007.java
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// TMP007
// This code is designed to work with the TMP007_I2CS I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/content/Temperature?sku=TMP007_I2CS#tabs-0-product_tabset-2

import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class TMP007
{
	public static void main(String args[]) throws Exception
	{
		// Create I2CBus
		I2CBus bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, TMP007 I2C address is 0x41(64)
		I2CDevice device = bus.getDevice(0x41);

		// Select configuration register
		// Continuous conversion, comparator mode
		byte[] config = {0x15, 0x40};
		device.write(0x02, config, 0, 2);

		// Read 2 bytes of data from address 0x03(3)
		// temp msb, temp lsb
		byte[] data = new byte[2];
		device.read(0x03, data, 0, 2);

		// Convert the data to 14-bits
		int temp = (((data[0] & 0xFF) * 256 + (data[1] & 0xFC)) / 4);
		if(temp > 8191)
		{
			temp -= 16384;
		}
		double cTemp = temp * 0.03125;
		double fTemp = cTemp * 1.8 + 32;

		// Output data to screen
		System.out.printf("Temperature in Celsius : %.2f C %n", cTemp);
		System.out.printf("Temperature in Fahrenheit : %.2f C %n", fTemp);
    }
}

File: Python/TMP007.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# TMP007
# This code is designed to work with the TMP007_I2CS I2C Mini Module available from ControlEverything.com.
# https://www.controleverything.com/content/Temperature?sku=TMP007_I2CS#tabs-0-product_tabset-2

import smbus
import time

# Get I2C bus
bus = smbus.SMBus(1)

# TMP007 address, 0x40(64)
# Select configuration register, 0x02(02)
#		0x1540(5440)	Continuous Conversion mode, Comparator mode
data = [0x1540]
bus.write_i2c_block_data(0x40, 0x02, data)

time.sleep(0.5)

# TMP007 address, 0x40(64)
# Read data back from 0x03(03), 2 bytes
# cTemp MSB, cTemp LSB
data = bus.read_i2c_block_data(0x40, 0x03, 2)

# Convert the data to 14-bits
cTemp = ((data[0] * 256 + (data[1] & 0xFC)) / 4) * 0.03125
if cTemp > 8191 :
	cTemp -= 16384
fTemp = cTemp * 1.8 + 32

# Output data to screen
print "Object Temperature in Celsius : %.2f C" %cTemp
print "Object Temperature in Fahrenheit : %.2f F" %fTemp
